### PR TITLE
Add an _Image protocol to tkinter.

### PR DESCRIPTION
**Pull request by rchen152**
I'm in the process of pulling a newer version of typeshed into Google
and have come across new type errors in code that passes
PIL.ImageTk.PhotoImage objects to tkinter.Label.configure().
Changing the method to instead accept a protocol fixes this.

I created the protocol by checking what attributes and methods are
shared between tkinter.Image and PIL.ImageTk.PhotoImage
(https://github.com/python-pillow/Pillow/blob/0bb6a19cbf250867e708c3c09aa68e857621d272/src/PIL/ImageTk.py#L65).

### CODE REVIEW

**Review by srittau — APPROVED** [review]
+1 for using a protocol, LGTM as is, but I'd appreciate it if you could add return types to the protocol.

File: stdlib/3/tkinter/__init__.pyi [hunk 1]
```diff
@@ -2831,6 +2831,12 @@ class OptionMenu(Menubutton):
     # configure, config, cget are inherited from Menubutton
     # destroy and __getitem__ are overrided, signature does not change
 
+class _Image(Protocol):
+    tk: _tkinter.TkappType
+    def __del__(self) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/tkinter/__init__.pyi:2836]: `__del__` being part of a protocol seems odd, does tk really require it?

Comment (rchen152) [reply stdlib/3/tkinter/__init__.pyi:2836]: I...don't really know. If there's documentation anywhere on what an image class should define to be tkinter-compatible, I haven't been able to find it. I just included all the methods that are present on both tkinter.Image and PIL.ImageTk.PhotoImage (except `__str__`).

**Comment by Akuli** [comment]
I think `__del__` is not necessary. Tkinter uses it to delete the underlying Tcl image, but there's no reason to require every tkinter-compatible image to have it.

Every tkinter image has to have a meaningful `__str__`, which gets used in the Tcl call that `configure()` does. But I don't know what including `__str__` in a protocol would do. For this purpose, it would be best to reject any objects that don't have a custom `__str__`, because that's very likely a programming error.

**Comment by srittau** [comment]
I assume that it wouldn't do anything, since every class has a `__str__` inherited from `object`.

**Comment by rchen152** [comment]
Sounds like I should remove `__del__` from the protocol then, thanks!

**Pull request merged**